src/solver/conv_asm_implicit_gemm_wrw_gtc_dynamic_xdlops.cpp

```diff
@@ -544,31 +878,76 @@ ConvAsmImplicitGemmGTCDynamicWrwXdlops::GetSolution(const ConvolutionContext& ct
     result.construction_params.push_back(kernel);
 
     const auto& conv_problem = ctx.conv_problem;
+    const auto& lowp_quant   = ctx.conv_problem.GetConv().lowp_quant;
 
-    result.invoker_factory = [conv_problem,
-                              log2_gemm_k_global_splits](const std::vector<Kernel>& kernels) {
+    auto opShapeArgs = ComputeDynamicIGemmWrwKernelArgs(conv_problem, log2_gemm_k_global_splits);
+
+    result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
         return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
             decltype(auto) data_ctx = primitive_parameters.CastTo<conv::WrWInvokeParams>();
             const auto& tensors     = data_ctx.tensors;
-            std::vector<KernelInvoke> ks;
-            std::transform(kernels.begin(),
-                           kernels.end(),
-                           std::back_inserter(ks),
-                           [&](const Kernel& k_wrw) { return handle.Run(k_wrw); });
+            const auto k            = handle.Run(kernels[0]);
+
+            std::vector<OpKernelArg> opArgs;
+            opArgs.reserve(3 + opShapeArgs.size()); // Avoids vector resize.
+            opArgs.emplace_back(tensors.x);
+            opArgs.emplace_back(tensors.dw);
+            opArgs.emplace_back(tensors.dy);
+
+            std::transform(opShapeArgs.begin(),
+                           opShapeArgs.end(),
+                           std::back_inserter(opArgs),
+                           [](const OpKernelArg& arg) { return arg; });
+
             float elapsed = 0;
             float zero    = 0.f;
 
-            SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
-            if(handle.IsProfilingEnabled())
-                elapsed += handle.GetKernelTime();
-
-            elapsed += CallImplicitGemmWrwDynamic(handle,
-                                                  conv_problem,
-                                                  tensors.x,
-                                                  tensors.dy,
-                                                  tensors.dw,
-                                                  ks,
-                                                  log2_gemm_k_global_splits);
+            if(conv_problem.IsFp32())
+            {
+                SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
+                if(handle.IsProfilingEnabled())
+                    elapsed += handle.GetKernelTime();
+
+                k(opArgs);
+                if(handle.IsProfilingEnabled())
+                    elapsed += handle.GetKernelTime();
+            }
+            else if(conv_problem.IsFp16())
+            {
+                if(log2_gemm_k_global_splits > 0)
+                {
+                    const auto& workSpace = data_ctx.workSpace;
```

Comment:
Please check workspace size when you use it, like this:
```C++
if(invoke_params.workSpaceSize < workspaceSize)
    MIOPEN_THROW("Not enough workspace for invoker of ConvAsmImplicitGemmGTCDynamicWrwXdlops");
```
This is a general rule for Invokers.

Reply:
Done, please check 1c191a0. @atamazov